src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -5506,6 +5513,28 @@ public Expression GenerateCallContains(Expression lhs, Expression rhs, bool igno
                 rhs.Cast(typeof(object)));
         }
 
+        public Expression GenerateCallAny(Expression lhs, Expression rhs)
+        {
+            return Expression.Call(
+                CachedReflectionInfo.ParserOps_AnyOperator,
+                s_executionContextParameter,
+                Expression.Constant(CallSite<Func<CallSite, object, IEnumerator>>.Create(PSEnumerableBinder.Get())),
+                Expression.Constant(CallSite<Func<CallSite, object, object, object>>.Create(PSBinaryOperationBinder.Get(ExpressionType.Equal, true, true))),
+                lhs.Cast(typeof(object)),
+                rhs.Cast(typeof(object)));
```

Comment:
I'm not sure how much it'd help here, but perhaps naming a few of the arguments here might make this method call more digestible.

Reply:
Not sure it would - the parameters are `(methodInfo, arg0, arg1, arg2, arg3, arg4)`

Reply:
> Not sure it would - the parameters are `(methodInfo, arg0, arg1, arg2, arg3, arg4)`

I've seen some folks use something like:

```csharp
    /* context: */ s_executionContextParameter,
    /* getEnumeratorSite: */ Expression.Constant(CallSite<Func<CallSite, object, IEnumerator>>.Create(PSEnumerableBinder.Get())),
    /* comparerSite: */ Expression.Constant(CallSite<Func<CallSite, object, object, object>>.Create(PSBinaryOperationBinder.Get(ExpressionType.Equal, true, true))),
    /* left: */ lhs.Cast(typeof(object)),
    /* right: */ rhs.Cast(typeof(object))
```

But since they're not references, they have a decent probability of lying later down the road :/